### ISSUE TITLE
Packer 0.7.2 is failing on response from vagrantcloud post-processor

### ISSUE THREAD

**Issue by bruno** [mitchellh/packer#1735]
When trying to create a new Virtualbox and push it to vagrant-cloud (note: this worked 20 days ago), I am getting the following:

```
[...]
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Running post-processor: vagrant
==> virtualbox-iso (vagrant): Creating Vagrant box for 'virtualbox' provider
    virtualbox-iso (vagrant): Copying from artifact: output-virtualbox-iso/packer-virtualbox-iso-1418095977-disk1.vmdk
    virtualbox-iso (vagrant): Copying from artifact: output-virtualbox-iso/packer-virtualbox-iso-1418095977.ovf
    virtualbox-iso (vagrant): Renaming the OVF to box.ovf...
    virtualbox-iso (vagrant): Compressing: Vagrantfile
    virtualbox-iso (vagrant): Compressing: box.ovf
    virtualbox-iso (vagrant): Compressing: metadata.json
    virtualbox-iso (vagrant): Compressing: packer-virtualbox-iso-1418095977-disk1.vmdk
==> virtualbox-iso: Running post-processor: vagrant-cloud
==> virtualbox-iso (vagrant-cloud): Verifying box is accessible: waysact/trusty64
Build 'virtualbox-iso' errored: 1 error(s) occurred:

* Post-processor failed: Error parsing box response: json: cannot unmarshal string into Go value of type uint
```

This is the configuration for our packer build

```
  "post-processors": [
    [{
      "type": "vagrant",
      "compression_level": 9,
      "output": "{{user `output_dir`}}/{{user `os_codename`}}{{user `arch`}}.box",
      "only": [
        "virtualbox-iso"
      ]
    },
    {
      "type": "vagrant-cloud",
      "box_tag": "waysact/{{user `os_codename`}}{{user `arch`}}",
      "access_token": "{{user `cloud_token`}}",
      "version": "{{user `version`}}",
      "only": [
        "virtualbox-iso"
      ]
    }]
```

We have a few versions already on vagrant-cloud, and pushing a box to vagrant cloud was working up until 20 days ago (roughly when I did the last `packer build`). The only thing that has changed in between is the `{{user 'version'}}` which is incremented (since we do `apt-get upgrade` etc ).

**Comment by ajw0100** [comment]
Any idea when this will be released? I'm getting the same error on Atlas and I can't checkout ^HEAD there.

**Comment by grafjo** [comment]
@ajw0100 maybe ask somebody on irc / freenode

**Comment by zspencer** [comment]
This is also blocking me. Would be great if a new release came out soon.

**Comment by jhulten** [comment]
Also blocking here.

**Comment by sethvargo** [comment]
Relates to https://github.com/mitchellh/packer/issues/1815

**Comment by acoulton** [comment]
Is there any workaround for this, or is the vagrant cloud post-processor totally broken at the moment?

**Comment by grafjo** [comment]
@acoulton you have to build packer by your own with https://github.com/mitchellh/packer/pull/1789

**Comment by acoulton** [comment]
@grafjo OK, thanks for the reply. I'll probably just wait for the next release then, I don't have a go compiler etc set up.

**Comment by pearkes** [comment]
If you use the currently released (as of this comment) Packer 0.7.5 you can use the Atlas post-processor and do something like this:

```json
    "post-processors": [
        [{
            "type": "vagrant",
            "keep_input_artifact": false
        },
        {
            "type": "atlas",
            "only": ["vmware-iso"],
            "artifact": "{{user `atlas_username`}}/{{user `atlas_name`}}",
            "artifact_type": "vagrant.box",
            "metadata": {
                "provider": "vmware_desktop",
                "version": "0.0.1"
            }
        },
        {
            "type": "atlas",
            "only": ["virtualbox-iso"],
            "artifact": "{{user `atlas_username`}}/{{user `atlas_name`}}",
            "artifact_type": "vagrant.box",
            "metadata": {
                "provider": "virtualbox",
                "version": "0.0.1"
            }
        }]
    ]
```

This is also fixed in HEAD for the Vagrant Cloud post-processor which will also still work. Just as a sidenote, the Vagrant Cloud post-processor uses a different API then the above Atlas post-processor, which uses a more generic artifact API. Both result in a Vagrant Boxes being created in Atlas.

Either way, I think we can safely call this resolved, let me know if you're still seeing issues.

**Comment by qm78** [comment]
With currently released packer 0.7.5, neither atlas nor vagrant post processor work for me to upload my virtualbox artifact to atlas. 

atlas post processor fails with: 
* Post-processor failed: Error uploading: client: 422 Unprocessable Entity

=> according to https://github.com/mitchellh/packer/issues/1986 this seems to be solved in master
=> proposed post-processor configuration by @pearkes still leads to: 422 Unprocessable Entity error

vagrantcloud post processor fails - as stated above - with:
* Post-processor failed: Error parsing box response: json: cannot unmarshal string into Go value of type uint

Stuck right now. Any ETA for the next release? Wishfully awaiting it :pray:
